engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/interop/generic/GetExecutableNameNode.java

```diff
@@ -25,6 +25,14 @@ public class GetExecutableNameNode extends Node {
   private final BranchProfile err = BranchProfile.create();
 
   Text execute(Object function) {
+    // Workaround for https://github.com/oracle/graal/issues/7359
+    if (!functionsLibrary.hasExecutableName(function)) {
+      err.enter();
+      Builtins builtins = EnsoContext.get(this).getBuiltins();
+      throw new PanicException(
+          builtins.error().makeTypeError(builtins.function(), function, "function"), this);
+    }
```

Comment:
Here's a, possibly temporary, workaround for https://github.com/oracle/graal/issues/7359

Essentially, it is possible that `getExecutableName` will not throw but still return `null`, so a compiler error on `null` will crash the interpreter.